pkg/3scale/amp/operator/upgrade.go

```diff
@@ -49,7 +49,7 @@ func (u *UpgradeApiManager) Upgrade() (reconcile.Result, error) {
 		return res, err
 	}
 
-	if u.Cr.Spec.System.FileStorageSpec != nil && u.Cr.Spec.System.FileStorageSpec.S3 != nil {
+	if u.Cr.Spec.System.FileStorageSpec != nil && (u.Cr.Spec.System.FileStorageSpec.DeprecatedS3 != nil || u.Cr.Spec.System.FileStorageSpec.S3 != nil) {
```

Comment:
Before upgrading, there is no way the user has filled *both* fields 
* `spec.system.fileStorage.amazonSimpleStorageService`
* `spec.system.fileStorage.simpleStorageService`

The upgrading procedure of S3 have some steps and the last one should be updating apimanager CR. Then you do not need to check both fields existence at the same time

Reply:
There is no way the user has filled both fields, but it might have one of the two set depending on the stage of the upgrade procedure that is being executed at one point. This is checking that either of them is set, not both of them at the same time

Reply:
When updating S3 section of the CR is the last step of the upgrading S3 procedure (like it is now and as should be), this change is just allowing entering upgrading S3 process one more time and not doing anything, as everything has already been upgraded. Am I right?

Reply:
You are right.
I think it's not a bad idea to allow re-entering after all the changes to ensure the current state is the desired state at all points in time, after having applied all changes, like you do with reconciliation.
I'll change it anyway to simplify the code a little bit anyway :+1:

Reply:
updating s3 section of the CR should be the last step. Would be good to add a comment to prevent anyone adding more steps after.

Reply:
Comment added :+1: